_posts/2023-07-07-announcing-ballerina-2201.7.0-swan-lake-update-7.md

```diff
@@ -0,0 +1,107 @@
+
+By leveraging GraalVM native images, Ballerina programs can achieve performance improvements and reduced startup times compared to traditional Java Virtual Machine(JVM) execution. This is particularly beneficial for applications with strict latency requirements or resource-constrained environments.
+You can generate a GraalVM native executable for your Ballerina project by executing the command below.
+
+```
+$ bal build --graalvm
+```
+
+The `--graalvm` flag triggers the GraalVM native image builder, which compiles the Ballerina code and its dependencies into a single, self-contained executable, which contains the modules in the current package, their dependencies, Ballerina runtime, and statically linked native code from the JDK.
+
+For more information on the necessary steps and additional considerations to ensure a successful compilation process of building a native executable with Ballerina and GraalVM, see [Build a GraalVM executable](https://ballerina.io/learn/graalvm-executable-overview/). 
+
+## Aggregation and grouping
+
+The language now supports the `group by` and `collect` clauses to perform aggregation and grouping-related operations as shown in the example below. 
+
+### Aggregation
+
+The `collect` clause categorizes a collection into one group as shown in the example below.
+
+```ballerina
+import ballerina/io;
+
+public function main() returns error? {
+    var orders = [
+        {orderId: 1, itemName: "Rich Dad Poor Dad", price: 23.4, quantity: 2},
+        {orderId: 1, itemName: "Rich Dad Poor Dad", price: 20.4, quantity: 1},
+        {orderId: 2, itemName: "Becoming", price: 21.5, quantity: 3},
+        {orderId: 1, itemName: "Becoming", price: 21.5, quantity: 3}
+    ];
+
+    var income = from var {price, quantity} in orders
+        let var totPrice = price * quantity
+        // The `collect` clause creates a single group and all variables become
+        // non-grouping keys.
+        collect sum(totPrice);
+
+    // Calculate the total income from all the orders.
+    io:println(income); // 196.2
+}
+```
+
+For more information on the usage of aggregation, see [Aggregation](https://ballerina.io/learn/work-with-data-using-queries-in-ballerina/#aggregation).
+
+### Grouping
+
+The `group by` clause groups a collection based on a `grouping-key`, which will be unique for each group as shown in the example below.
+
+```ballerina
+import ballerina/io;
+
+public function main() returns error? {
+    var orders = [
+        {orderId: 1, itemName: "Rich Dad Poor Dad", price: 23.4, quantity: 2},
+        {orderId: 1, itemName: "Rich Dad Poor Dad", price: 20.4, quantity: 1},
+        {orderId: 2, itemName: "Becoming", price: 21.5, quantity: 3},
+        {orderId: 1, itemName: "Becoming", price: 21.5, quantity: 3}
+    ];
+
+    var items = from var {orderId, itemName} in orders
```

Comment:
Here the LHS type is a map, but it is not clear from the code. Please use types instead of `var` here as well.

Reply:
Updated.